List custom drawn control

DenisUngemach opened the pull request:
Fixes: https://github.com/swt-initiative31/prototype-skija/issues/19

DenisUngemach commented:
- I think it is ok, that the fonts are different. 
- The background could be changed.


It seems the listeners trigger are missing. 
The selection in the scrollbars does not seems to work.
And the computeSize method should return the sum of the textheights of each line and the max width of the lines.

@praveen-skp 
Can you checkout the branch list directly and commit the changes?
We can also do this on monday.

<img width="440" alt="image" src="https://github.com/user-attachments/assets/44db1646-fc4e-4172-b9c5-4c9162dd41cd" />

praveen-skp commented:
@DenisUngemach , sure i can do the changes in the list branch.

DenisUngemach commented:
Can you fix the tests:

For more details on these failures and errors, see [this check](https://github.com/swt-initiative31/prototype-skija/runs/37757887983).

tmssngr commented:
IMHO following is still pending
- base to master
- the whole line needs to appear selected
- the Size selection is ignored (ControlExample)
- the disabled state is not rendered
- pressing left mouse button over one row, moving it to another and releasing it there selects the latter (the native widgets changes the selection while moving)
- I can't tell something about scrolling
- introduce `ListRenderer` and `DefaultListRenderer`

Maybe it would be better to first convert the Scrollable to a CustomScrollable?

DenisUngemach commented:
Thanks Thomas, i will check these.

DenisUngemach commented:
> IMHO following is still pending
> 
> * rebase to master
> * the whole line needs to appear selected
> * the Size selection is ignored (ControlExample)
> * the disabled state is not rendered
> * pressing left mouse button over one row, moving it to another and releasing it there selects the latter (the native widgets changes the selection while moving)
> * I can't tell something about scrolling
> * introduce `ListRenderer` and `DefaultListRenderer`
> 
> Maybe it would be better to first convert the Scrollable to a CustomScrollable?

I have fixed them. And updated the code. 
DefaultListRenderer also exists now.

DenisUngemach commented:
> * I can't tell something about scrolling

This i did not fix :-).
